Src/Plugins/Security/SecurePlaintextSecretsValidators/SEC101_003.GoogleApiKeyValidator.cs

```diff
@@ -3,113 +3,148 @@
-        {
-            FlexMatch secret = groups["secret"];
+        internal const string EndpointUriTemplate = "https://maps.googleapis.com/maps/api/directions/json?key={0}&origin=Seattle&destination=Redmond&units=metric&language=en&mode=driving";
 
+        protected override string Platform => nameof(AssetPlatform.Google);
+
+        protected override bool IsFalsePositiveOrBelongsToOtherSecurityModel(string secret)
+        {
             // It is highly likely we do not have a key if we can't
             // find at least one letter and digit within the pattern.
-            if (!secret.Value.ToString().ContainsDigitAndLetter())
-            {
-                return ValidationResult.CreateNoMatch();
-            }
-
-            var validationResult = new ValidationResult
-            {
-                Fingerprint = new Fingerprint
-                {
-                    Secret = secret.Value,
-                    Platform = nameof(AssetPlatform.Google),
-                },
-            };
-
-            return new[] { validationResult };
+            return !secret.ContainsDigitAndLetter();
         }
 
         protected override ValidationState IsValidDynamicHelper(ref Fingerprint fingerprint,
                                                                 ref string message,
                                                                 IDictionary<string, string> options,
                                                                 ref ResultLevelKind resultLevelKind)
         {
-            const string Invalid = "RequestDenied: The provided API key is invalid";
-            const string Expired = "RequestDenied: The provided API key is expired";
-            const string RevokedKey = "RequestDenied: Google has disabled the use of APIs from this API project";
-            const string ProjectNotAuthorized = "RequestDenied: This API project is not authorized to use this API";
-            const string KeyNotAuthorized = "RequestDenied: This API key is not authorized to use this service or API";
-            const string RefererRestrictions = "RequestDenied: API keys with referer restrictions cannot be used with this API";
-            const string IPNotAuthorized = "RequestDenied: This IP, site or mobile application is not authorized to use this API key";
-            const string EnableBilling = "RequestDenied: You must enable Billing on the Google Cloud Project at https://console.cloud.google.com/project/_/billing/enable Learn more at https://developers.google.com/maps/gmp-get-started";
-            const string Deleted = "RequestDenied: This API project was not found. This API project may have been deleted or may not be authorized to use this API. You may need to enable the API under APIs in the console";
+            const string RequestDeniedErrorCode = "REQUEST_DENIED";
+            const string Invalid = "The provided API key is invalid";
```

Comment:
>Invalid [](http://example.com/codeflow?start=25&length=7)

Can we operate against results codes not string messages? or no?

Reply:
All these failed cases have same error code "REQUEST_DENIED", but different error messages.

When use Google library, the error message is consist of error code + error message.